### PR TITLE
Issue #61 create CarDao

### LINKED ISSUE #61: Create CarDao

### PR DESCRIPTION
**Pull request by JustRevolt**

### CODE REVIEW

**Review by nikaalin — APPROVED** [review]

File: src/main/java/com/ifmo/epampractice/daoimpl/CarDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import com.ifmo.epampractice.dao.DBConnectorPostgres;
+import com.ifmo.epampractice.entity.CarEntity;
+import com.ifmo.epampractice.enums.CarStatus;
+
+import java.sql.*;
+import java.util.LinkedList;
+import java.util.List;
+
+public class CarDaoImpl implements CarDao {
+    DBConnectorPostgres dbConnector = DBConnectorPostgres.getInstance();
```
Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/daoimpl/CarDaoImpl.java]: could be private?

File: src/main/java/com/ifmo/epampractice/daoimpl/CarDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+    @Override
+    public List<CarEntity> getAll() {
+        List<CarEntity> carEntityList = new LinkedList<>();
+        final String sql = "SELECT * FROM car";
```
Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/daoimpl/CarDaoImpl.java]: Move constants to class level

File: src/main/java/com/ifmo/epampractice/daoimpl/CarDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return carEntityList;
+    }
+
+    @Override
+    public CarEntity get(int id) {
+        final String sql = "SELECT * FROM car WHERE id = ?";
```
Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/daoimpl/CarDaoImpl.java]: Move constants to class level

**Review by a-deriabin — APPROVED** [review]

**Review by a-deriabin — APPROVED** [review]

File: src/main/java/com/ifmo/epampractice/daoimpl/CarDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.LinkedList;
+import java.util.List;
+
+public class CarDaoImpl implements CarDao {
+    private final String GET_ALL_QUERY = "SELECT * FROM car";
```
Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/daoimpl/CarDaoImpl.java]: constants should be private static final

File: src/main/java/com/ifmo/epampractice/daoimpl/CarDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+            "reg_number = ?, vin_number = ?, status = ?::e_status_car WHERE id = ?";
+    private final String DELETE_QUERY = "DELETE FROM car WHERE id = ?";
+    private final String SAVE_QUERY = "INSERT INTO car (model_id, color, reg_number, " +
+            "vin_number, status) VALUES (?, ?, ?, ?, ?::e_status_car)";
+
+    private DBConnectorPostgres dbConnector = DBConnectorPostgres.getInstance();
+
+    @Override
+    public List<CarEntity> getAll() {
+        List<CarEntity> carEntityList = new LinkedList<>();
```
Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/daoimpl/CarDaoImpl.java]: Why LinkedList?

**Review by a-deriabin — CHANGES_REQUESTED** [review]

**Review by a-deriabin — APPROVED** [review]

**Review by PavelEzhkov — APPROVED** [review]

**Pull request merged**